django/db/models/sql/query.py

```diff
@@ -972,8 +972,19 @@ def add_annotation(self, annotation, alias, is_summary=False):
         self.append_annotation_mask([alias])
         self.annotations[alias] = annotation
 
-    def _prepare_as_filter_value(self):
-        return self.clone()
+    def resolve_expression(self, query, *args, **kwargs):
+        clone = self.clone()
+        # Subqueries need to use a different set of aliases than the
```

Comment:
Alright, I thought it ended up being 81-82 chars with `query` and wanted to avoid orphaning it on a second line so I pushed `outer` to the new line as well.